### ISSUE TITLE
DefaultDatabaseResolver null when taking backups

### ISSUE THREAD

**Issue by thordal-indykite** [neo4j/neo4j#13228]
When running backups of multiple databases at the same time an exception is thrown on the backup server.

**Neo4j Version:** Enterprise 5.5.0
**Dockerfile base:**
```
FROM debian:bullseye-slim
ENV JAVA_HOME=/opt/java/openjdk
COPY --from=eclipse-temurin:17 $JAVA_HOME $JAVA_HOME
```
**Cloud provider**: Google Cloud

The found exception is 
```
Exception in inbound for channel: [id: 0x1a0667b3, L:/10.7.11.3:6362 - R:/10.7.6.184:58360] java.lang.NullPointerException: Cannot invoke "org.neo4j.kernel.database.DefaultDatabaseResolver.defaultDatabase(String)" because "defaultDatabaseResolver" is null
  at com.neo4j.server.security.enterprise.systemgraph.versions.SupportedEnterpriseSecurityComponentVersion.lambda$getRelevantRolesAndPrivileges$7(SupportedEnterpriseSecurityComponentVersion.java:285) ~[neo4j-security-enterprise-5.6.0.jar:5.6.0]
  at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
  at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179) ~[?:?]
  at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
  at java.util.Iterator.forEachRemaining(Iterator.java:133) ~[?:?]
  at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845) ~[?:?]
  at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
  at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
  at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?]
  at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
  at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[?:?]
  at com.neo4j.server.security.enterprise.systemgraph.versions.SupportedEnterpriseSecurityComponentVersion.getRelevantRolesAndPrivileges(SupportedEnterpriseSecurityComponentVersion.java:286) ~[neo4j-security-enterprise-5.6.0.jar:5.6.0]
  at com.neo4j.server.security.enterprise.systemgraph.versions.SupportedEnterpriseSecurityComponentVersion.getBackupCommands(SupportedEnterpriseSecurityComponentVersion.java:258) ~[neo4j-security-enterprise-5.6.0.jar:5.6.0]
  at com.neo4j.server.security.enterprise.systemgraph.EnterpriseSecurityGraphComponent.getBackupCommands(EnterpriseSecurityGraphComponent.java:182) ~[neo4j-security-enterprise-5.6.0.jar:5.6.0]
  at com.neo4j.causalclustering.dbms.protocol.messages.get_meta_data.GetMetadataRequestHandler.lambda$getBackupCommands$0(GetMetadataRequestHandler.java:57) ~[neo4j-protocol-dbms-5.6.0.jar:5.6.0]
  at java.util.Optional.map(Optional.java:260) ~[?:?]
  at com.neo4j.causalclustering.dbms.protocol.messages.get_meta_data.GetMetadataRequestHandler.getBackupCommands(GetMetadataRequestHandler.java:53) ~[neo4j-protocol-dbms-5.6.0.jar:5.6.0]
  at com.neo4j.causalclustering.dbms.protocol.messages.get_meta_data.GetMetadataRequestHandler.channelRead0(GetMetadataRequestHandler.java:43) ~[neo4j-protocol-dbms-5.6.0.jar:5.6.0]
  at com.neo4j.causalclustering.dbms.protocol.messages.get_meta_data.GetMetadataRequestHandler.channelRead0(GetMetadataRequestHandler.java:22) ~[neo4j-protocol-dbms-5.6.0.jar:5.6.0]
  at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:102) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:102) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:102) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at com.neo4j.causalclustering.net.app.server.ServerInboundRequestPublisher.channelRead0(ServerInboundRequestPublisher.java:27) ~[neo4j-cluster-common-5.6.0.jar:5.6.0]
  at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346) ~[netty-codec-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318) ~[netty-codec-4.1.87.Final.jar:4.1.87.Final]
  at com.neo4j.causalclustering.net.app.common.RequestDecoderDispatcher.channelRead(RequestDecoderDispatcher.java:39) ~[neo4j-cluster-common-5.6.0.jar:5.6.0]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at com.neo4j.causalclustering.net.app.common.MessageTypeHandler.channelRead(MessageTypeHandler.java:39) ~[neo4j-cluster-common-5.6.0.jar:5.6.0]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346) ~[netty-codec-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318) ~[netty-codec-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at com.neo4j.causalclustering.protocol.NettyPipelineBuilder$ErrorHandlerHead.channelRead(NettyPipelineBuilder.java:274) ~[neo4j-cluster-common-5.6.0.jar:5.6.0]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:800) ~[netty-transport-classes-epoll-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:499) ~[netty-transport-classes-epoll-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:397) ~[netty-transport-classes-epoll-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.87.Final.jar:4.1.87.Final]
  at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
  at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
  at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.87.Final.jar:4.1.87.Final]
  at java.lang.Thread.run(Thread.java:833) ~[?:?]
","logging.googleapis.com/sourceLocation":{"function":"org.neo4j.logging.internal.PrefixedLog.error"},"logging.googleapis.com/insertId":"1812826","_exception":{"class":"java.lang.NullPointerException","message":"Cannot invoke "org.neo4j.kernel.database.DefaultDatabaseResolver.defaultDatabase(String)" because "defaultDatabaseResolver" is null","stackTrace":"java.lang.NullPointerException: Cannot invoke "org.neo4j.kernel.database.DefaultDatabaseResolver.defaultDatabase(String)" because "defaultDatabaseResolver" is null
  at com.neo4j.server.security.enterprise.systemgraph.versions.SupportedEnterpriseSecurityComponentVersion.lambda$getRelevantRolesAndPrivileges$7(SupportedEnterpriseSecurityComponentVersion.java:285) ~[neo4j-security-enterprise-5.6.0.jar:5.6.0]
  at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
  at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179) ~[?:?]
  at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
  at java.util.Iterator.forEachRemaining(Iterator.java:133) ~[?:?]
  at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845) ~[?:?]
  at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
  at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
  at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?]
  at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
  at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[?:?]
  at com.neo4j.server.security.enterprise.systemgraph.versions.SupportedEnterpriseSecurityComponentVersion.getRelevantRolesAndPrivileges(SupportedEnterpriseSecurityComponentVersion.java:286) ~[neo4j-security-enterprise-5.6.0.jar:5.6.0]
  at com.neo4j.server.security.enterprise.systemgraph.versions.SupportedEnterpriseSecurityComponentVersion.getBackupCommands(SupportedEnterpriseSecurityComponentVersion.java:258) ~[neo4j-security-enterprise-5.6.0.jar:5.6.0]
  at com.neo4j.server.security.enterprise.systemgraph.EnterpriseSecurityGraphComponent.getBackupCommands(EnterpriseSecurityGraphComponent.java:182) ~[neo4j-security-enterprise-5.6.0.jar:5.6.0]
  at com.neo4j.causalclustering.dbms.protocol.messages.get_meta_data.GetMetadataRequestHandler.lambda$getBackupCommands$0(GetMetadataRequestHandler.java:57) ~[neo4j-protocol-dbms-5.6.0.jar:5.6.0]
  at java.util.Optional.map(Optional.java:260) ~[?:?]
  at com.neo4j.causalclustering.dbms.protocol.messages.get_meta_data.GetMetadataRequestHandler.getBackupCommands(GetMetadataRequestHandler.java:53) ~[neo4j-protocol-dbms-5.6.0.jar:5.6.0]
  at com.neo4j.causalclustering.dbms.protocol.messages.get_meta_data.GetMetadataRequestHandler.channelRead0(GetMetadataRequestHandler.java:43) ~[neo4j-protocol-dbms-5.6.0.jar:5.6.0]
  at com.neo4j.causalclustering.dbms.protocol.messages.get_meta_data.GetMetadataRequestHandler.channelRead0(GetMetadataRequestHandler.java:22) ~[neo4j-protocol-dbms-5.6.0.jar:5.6.0]
  at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:102) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:102) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:102) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at com.neo4j.causalclustering.net.app.server.ServerInboundRequestPublisher.channelRead0(ServerInboundRequestPublisher.java:27) ~[neo4j-cluster-common-5.6.0.jar:5.6.0]
  at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346) ~[netty-codec-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318) ~[netty-codec-4.1.87.Final.jar:4.1.87.Final]
  at com.neo4j.causalclustering.net.app.common.RequestDecoderDispatcher.channelRead(RequestDecoderDispatcher.java:39) ~[neo4j-cluster-common-5.6.0.jar:5.6.0]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at com.neo4j.causalclustering.net.app.common.MessageTypeHandler.channelRead(MessageTypeHandler.java:39) ~[neo4j-cluster-common-5.6.0.jar:5.6.0]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346) ~[netty-codec-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318) ~[netty-codec-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at com.neo4j.causalclustering.protocol.NettyPipelineBuilder$ErrorHandlerHead.channelRead(NettyPipelineBuilder.java:274) ~[neo4j-cluster-common-5.6.0.jar:5.6.0]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:800) ~[netty-transport-classes-epoll-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:499) ~[netty-transport-classes-epoll-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:397) ~[netty-transport-classes-epoll-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.87.Final.jar:4.1.87.Final]
  at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.87.Final.jar:4.1.87.Final]
  at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
  at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
  at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.87.Final.jar:4.1.87.Final]
  at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

The backup being run is
```
  neo4j-admin database backup  \
    --to-path="/backups" \
    --from=$NEO4J_ADDR \
    --type=FULL \
    --verbose \
    --expand-commands=true \
    --include-metadata=all \
    --pagecache=$PAGE_CACHE \
    ikg*
```
with a page cache of 256M and NEO4J_ADDR set to the headless service for our DB cluster and the ikg* pattern matching ~100 DBs.

**Comment by mnd999** [comment]
Thanks for the report. I'm not sure what is going on here from your stack trace although I agree it does look as though something has gone wrong. I think I probably need a bit more information to look into it further:

* Is the server 5.5 or 5.6? The stack trace suggests 5.6.
* Are you running standalone or in a cluster?
* Do see any other errors in the logs? During server startup for example?
* Are you using the home database feature?
* Approximately how many backups are you running concurrently? Do they all start at exactly the same time?

**Comment by thordal-indykite** [comment]
- Is the server 5.5 or 5.6? The stack trace suggests 5.6.
- - The backup runs 5.5, the cluster 5.6
- Are you running standalone or in a cluster?
- - 3 core cluster
- Are you using the home database feature?
- - For users? I believe most just have the default `neo4j` as home
- Approximately how many backups are you running concurrently? Do they all start at exactly the same time?
- - About 70-80. We have a large amount of databases starting with `ikg` so we run the backup job with the database specified as `ikg*`

The backup job does log the warning:
```
"2023-06-28 09:22:33.551+0000 WARN  [c.n.c.s.StoreCopyJob] [ikg8de4e145bc31424f9ca4368fbc8afff9v1/00979e4c] [ikg8de4e145bc31424f9ca4368fbc8afff9v1/00979e4c] Failed to get store file 'neostore' from 'neo4j-5-headless.neo4j-5.svc.cluster.local:6362'. Response: E_DATABASE_UNKNOWN."
```
The database `ikg8de4e145bc31424f9ca4368fbc8afff9v1` cannot be found with `SHOW DATABASES` so I wonder why we try to back it up when running `neo4j-admin backup ... ikg*`?

**Comment by mnd999** [comment]
Thanks for that updated. I looked into it and discussed with my colleagues in the clustering team and it appears the issue is already resolved in Neo4j 5.7. I can't link you a commit unfortunately as the change is wholly within the enterprise codebase.
